src/openrct2/peep/Peep.cpp

```diff
@@ -3381,3 +3375,20 @@ bool Peep::HasItem(ShopItem peepItem) const
 {
     return GetItemFlags() & EnumToFlag(peepItem);
 }
+
+void Peep::SetDestination(const CoordsXY& coords)
+{
+    DestinationX = static_cast<uint16_t>(coords.x);
+    DestinationY = static_cast<uint16_t>(coords.y);
+}
+
+void Peep::SetDestination(const CoordsXY& coords, int32_t tolerance)
+{
+    SetDestination(coords);
+    DestinationTolerance = tolerance;
+}
+
+CoordsXY Peep::GetDestination() const
+{
+    return CoordsXY{ static_cast<int16_t>(DestinationX), static_cast<int16_t>(DestinationY) };
```

Comment:
Why did you add these casts?

Reply:
DestinationX and DestinationY are uint16_t, so explicit cast. Those fields will be eventually superseded by CoordsXY.

Reply:
But uint16_t fits into int32_t, so why does it need to be cast?

Reply:
Signed vs unsigned. But I think DestinationX and DestinationY never use negative numbers in that regard, I'll eliminate the casts.